addons/account/models/account_move_line.py

```diff
@@ -960,6 +964,56 @@ def _compute_all_tax(self):
                     'tax_tag_ids': [(6, 0, compute_all_currency['base_tags'])],
                 }
 
+    @api.depends('account_id', 'company_id')
+    def _compute_discount_allocation_key(self):
+        for line in self:
+            if line.display_type == 'discount':
+                line.discount_allocation_key = frozendict({
+                    'account_id': line.account_id.id,
+                    'move_id': line.move_id.id,
+                    'name': line.name,
+                })
+            else:
+                line.discount_allocation_key = False
+
+    @api.depends('account_id', 'company_id', 'discount', 'price_unit', 'quantity')
+    def _compute_discount_allocation_needed(self):
+        for line in self:
+            line.discount_allocation_dirty = True
+            line.discount_allocation_needed = False
+            discount_allocation_account = line.move_id.get_discount_allocation_account()
+
+            if not discount_allocation_account or line.display_type != 'product' or line.currency_id.is_zero(line.discount):
+                continue
+
+            discounted_amount_currency = line.quantity * line.price_unit * line.discount/100
+            discount_allocation_needed = {}
+            discount_allocation_needed_vals = discount_allocation_needed.setdefault(
+                frozendict({
+                    'move_id': line.move_id.id,
+                    'account_id': line.account_id.id,
+                    'display_type': 'discount',
```

Comment:
The keys of the fronzendict should be the same as the ones from `discount_allocation_key`

I know it isn't like that on some other sync functions (like for EPD), and that is a mistake (causing useless re-computations and consuming SQL ids)

Reply:
Thanks for the explanation. Indeed I wasn't sure if the both keys were supposed to contain the same fields. I moved the `display_type: discount` to the dict below, but I realized it also work without it (the created lines have the display_type set to `discount` without the need to specify it. Should I keep it there for explicitly indicating that it is this value we want or should I just remove it ?